assets/src/containers/IndexPage.js

```diff
@@ -35,19 +35,17 @@ function IndexPage (props) {
   return (
     <Grid container spacing={16} className={classes.root}>
       <Grid item xs={12} className={classes.container}>
-        {courseInfo
-          ? <Grid
-            container
-            className={classes.wrapper}
-            spacing={8}
-          >
-            {routes(courseId, activeCourseViews).map((props, key) =>
-              <Link style={{ textDecoration: 'none' }} to={props.path} key={key}>
-                <SelectCard cardData={props} />
-              </Link>
-            )}
-          </Grid>
-          : <Spinner />}
+        <Grid
```

Comment:
This is what you assume when a prop is passed down to further down we assume the value is fetched. But earlier behavior is not like that just rendering all the component when things are getting warmed up

Reply:
It should be okay since we're only rendering this component once `courseInfo` is loaded in App? Or does it need to be reverted?

Reply:
I don't know what you mean by reverting. The index page will show what all the views the course has so it relies on `courseInfo` data. if that courseInfo is not fetched index page won't load and spinner keeps on spinning.  

Another thing I realized is that in the `useFetch` we are assuming that data we are getting is correct and no error handling is done. 
 ```
fetch(dataURL, fetchOptions)
          .then(res => res.json())
          .then(data => {
            cache.set(dataURL, data)
            setData(data)
            setLoaded(true)
          })
```
If something went wrong in fetching the data the subsequest receipent of the `courseInfo` data just grab it as with out considering bad data as an example 
```
loaded
          ? <>
<DashboardAppBar
              onMenuBarClick={setSideDrawerState}
              sideDrawerState={sideDrawerState}
              user={user}
              **courseName={courseInfo.name}**
              courseId={courseId} />
```

Reply:
If `courseInfo` is not fetched, should we display an error message?

I'll add error handling to `useFetch`, maybe by having it return `[loaded, error, data]`. Every component that loads data will then have to check if the data loaded correctly, and if there's an error, display some error message to the user.

Reply:
When `App.js` don't get the `courseInfo` yes It needs to display error message app is not loaded for reasons.   `grades/assignments/files` views when making their content call error handling is done. None of the views `useFetch.js` (files might be using it now but needs to change to accommodate view disabled msg). But once good `courseInfo` data is fetched the following components like `index`, `DashboardAppBar`,`SideDrawer` etc can simply grab it. 

If the subsequents component can check but what is the need for it if we are catching it at higher up in the component tress

Reply:
Good idea, so if `courseInfo` isn't fetched correctly in `App.js`, then it'll just display an error message.

I'll take this one.

Reply:
if some error occurs the `/info` call is responding with `{}` if a course doesn't exist. This could happen if a course is not there in DB or DB connectivity problems. But it won't respond with a status code. The `courseInfo` json object has various parts to it that pulling info from different DB tables, I need to make sure all the error has the same error message for consistency sake. I think if it responds with a status code that it might be easier. Let me analyze what the backend call is doing so that Frontend can predict the format. Which do you think you prefer status code or `200 Ok with {}`

Reply:
OK, I have reviewed this use case. I think if `courseInfo` don't return valid data then the BE is returning with `{}`. This is also the case when user types in bad course number that is not part of the Myla participation then the message should say "Tool is not enabled for this course"

Reply:
That makes sense. If there's a network error or the server is down, then should the same message ("Tool is not enabled for this course") be shown, or should there be a different message shown?

Reply:
I think if you see the response as {} that means course data couldn't be fetched. So that message should be fine. But the `useFetch`  FetchAPI should handle when the promise fails, currently, the code only handles success case only.

Reply:
This should be resolved now.